src/Host/Storage.php

```diff
@@ -14,6 +14,73 @@
 
 class Storage
 {
+	/**
+	 * In a multi user environment where multiple projects are located on the same instance(s)
+	 *	if two or more users are deploying at the same time on the same environment, 
+	 *	the generated hosts files are overwritten or worse, the deploy will fail 
+	 *	due to file permissions
+	 *
+	 * Make the persistent storage folder configurable or use the system tmp 
+	 *	as default if not possible
+	 * 
+	 * @return string
+	 * @throws Exception
+	 */
+	private static function _getPersistentStorageLocation() {
+		$config = \Deployer\Deployer::get()->config;
+
+		// use the system temporary folder and the current pid as default 
+		// persistent storage in case we can't use the configured value
+		// or we can't create the default location
+		$tmp = sys_get_temp_dir() . '/' . posix_getpid();
+
+		// get the location for the deployer configuration
+		$deployerConfig = $config->has('deployer_config') ? $config->get('deployer_config') : null;
```

Comment:
It's over this PR, I think such functionality should done in separate PR.

Anyway I don't see way it's needed. Please remove everything related to deployer_config.